WordPress/Classes/Utility/WPTableViewHandler.m

```diff
@@ -69,6 +68,110 @@ - (void)clearCachedRowHeights
     [self.cachedRowHeights removeAllObjects];
 }
 
+- (void)refreshTableView
+{
+    [self clearCachedRowHeights];
+    [self.tableView reloadData];
+}
+
+- (void)refreshTableViewPreservingOffset
+{
+    [self preserveRowInfoBeforeContentChanges];
+
+    // Make sure its necessary to account for previously existing rows.
+    // We check here vs in `controllerWillChangeContent:` in order to reload
+    // the table view if necessary.
+    if ([self.fetchedResultsBeforeChange count] == 0) {
+        if ([self.delegate respondsToSelector:@selector(tableViewHandlerWillRefreshTableViewPreservingOffset:)]) {
+            [self.delegate tableViewHandlerWillRefreshTableViewPreservingOffset:self];
+        }
+
+        [self.tableView reloadData];
+        [self discardPreservedRowInfo];
+
+        if ([self.delegate respondsToSelector:@selector(tableViewHandlerDidRefreshTableViewPreservingOffset:)]) {
+            [self.delegate tableViewHandlerDidRefreshTableViewPreservingOffset:self];
+        }
+        return;
+    }
+
+    self.refreshingTableViewPreservingOffset = YES;
+
+    // Get the current visible index paths before reloading data.
+    NSArray *visibleIndexPaths = [self.tableView indexPathsForVisibleRows];
+
+    // Get the delta of the current offset and the sum of the preserved row
+    // heights, up to but not including the first visible row.
+    CGFloat offsetHeightDelta = [self contentOffsetAndPreservedRowHeightDelta];
+
+    // Notify the delegate that a refresh is about to occur. Allows the delegate
+    // perform any final clean up, e.g. dismissing a UIRefreshControl.
+    if ([self.delegate respondsToSelector:@selector(tableViewHandlerWillRefreshTableViewPreservingOffset:)]) {
+        [self.delegate tableViewHandlerWillRefreshTableViewPreservingOffset:self];
+    }
+
+    // Clear cached heights and reload the tableview
+    [self refreshTableView];
+
+    // Find the indexPath of the first visible object after changes
+    // If the first object was deleted, try the next in line of the previously
+    // visible objects until a match is found.
+    CGFloat heightAdjustment = 0;
+    NSIndexPath *newIndexPath = nil;
+    for (NSInteger i = 0; i < [visibleIndexPaths count]; i++) {
+        NSIndexPath *path = visibleIndexPaths[i];
+        NSInteger index = [self.fetchedResultsIndexPathsBeforeChange indexOfObject:path];
+        NSManagedObject *obj = self.fetchedResultsBeforeChange[index];
+        if (obj.isFault) {
+            NSError *error;
+            obj = [self.resultsController.managedObjectContext existingObjectWithID:obj.objectID error:&error];
+            if (error) {
+                DDLogError(@"%@, Error %@", NSStringFromSelector(_cmd), error);
+            }
```

Comment:
Actually no, we want it to pass through so we still add the row height below.  Nice thought tho!